src/controllers/module/event/eventController.ts

```diff
@@ -469,20 +509,63 @@ export class EventController {
 			const booking = this.findBooking(bookings, bookingId)
 
 			booking.status = Booking.Status.CONFIRMED
+			await this.learnerRecord.updateBooking(req.params.eventId, booking, '')
 
-			this.learnerRecord.updateBooking(req.params.eventId, booking)
-
-			res.redirect(
+			return res.redirect(
 				`/content-management/courses/${req.params.courseId}/modules/${req.params.moduleId}/events/${
 					req.params.eventId
 				}/attendee/${req.params.bookingId}`
 			)
 		}
 	}
 
-	public cancelEvent() {
-		return async (request: Request, response: Response) => {
-			response.render('page/course/module/events/cancel')
+	public getCancelBooking() {
+		return async (req: Request, res: Response) => {
+			const event = res.locals.event
+			const eventDateWithMonthAsText: string = DateTime.convertDate(event.dateRanges[0].date)
+
+			const bookings = await this.learnerRecord.getEventBookings(event.id)
+			const bookingId = req.params.bookingId
+			const booking = this.findBooking(bookings, bookingId)
+
+			res.render('page/course/module/events/cancel-attendee', {
+				booking: booking,
+				eventDateWithMonthAsText: eventDateWithMonthAsText,
+			})
+		}
+	}
+
+	public cancelBooking() {
+		return async (req: Request, res: Response) => {
+			const data = {
+				...req.body,
+			}
+
+			const bookings = await this.learnerRecord.getEventBookings(req.params.eventId)
+			const bookingId = req.params.bookingId
+			const booking = this.findBooking(bookings, bookingId)
+
+			if (data.reason == '') {
```

Comment:
To tidy this up a little, I'd be keen to use the validator annotation that @robleney-kainos introduced. You can see an example of this in `CourseController.ts`